src/axom/quest/MarchingCubes.hpp

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file MarchingCubes.hpp
+ *
+ * \brief Consists of classes implementing marching cubes algorithm to
+ * compute isocontour from a scalar field in a blueprint mesh.
+ */
+
+#ifndef AXOM_PRIMAL_MARCHINGCUBES_H_
+#define AXOM_PRIMAL_MARCHINGCUBES_H_
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/mint/mesh/UnstructuredMesh.hpp"
+
+// Conduit includes
+#include "conduit_node.hpp"
+
+// C++ includes
+#include <string>
+
+// Add some helper preprocessor defines for using OPENMP, CUDA, and HIP policies
+// within the marching cubes implementation.
+#if defined(AXOM_USE_RAJA)
```

Comment:
This really shows that we should probably rethink our original compiler defines. This is out of the scope of the PR but do we have any place in Axom that has with cuda but without raja/umpire?  Should those just be implied?